cpp/src/parquet/encoding.cc

```diff
@@ -2007,7 +2001,7 @@ class DictByteArrayDecoderImpl : public DictDecoderImpl<ByteArrayType>,
     // space for binary data.
     RETURN_NOT_OK(helper.Prepare());
 
-    auto dict_values = reinterpret_cast<const ByteArray*>(dictionary_->data());
+    const auto* dict_values = dictionary_->data_as<ByteArray>();
```

Comment:
Do we need to specify `const` and `*` explicitly here?

Reply:
No, `const auto* v = ...` is equal to `auto dict_values = ..`. It's just a style problem

Let me update it

Reply:
I like `auto*` because it communicates the assignment is just a pointer assignment and not some expensive copy/move operation. And it's creates a good symmetry with `auto&` that requires the `&` otherwise it becomes a copy.